presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -914,7 +916,9 @@ public PlanWithProperties visitSemiJoin(SemiJoinNode node, PreferredProperties p
                 else {
                     filteringSource = node.getFilteringSource().accept(this, PreferredProperties.partitionedWithNullsAndAnyReplicated(ImmutableSet.copyOf(filteringSourceVariables)));
 
-                    if (filteringSource.getProperties().isNodePartitionedOn(filteringSourceVariables, true) && !filteringSource.getProperties().isSingleNode()) {
+                    //Never use exact partitioning because if you do some tests fail because the partitioning doesn't match exactly and this code attempts to exchange
+                    //with null and any inputs. This a temp comment for review.
```

Comment:
Oh, it seems to be a temporary comment

Reply:
The comment is calling out the real issue where I have to specify exact partitioning false for this branch. I wish I had said which test it was that fails :-P